scipy/optimize/_direct.py

```diff
@@ -0,0 +1,203 @@
+    maxiter : int, optional
+        Maximum number of iterations. Default is 6000.
+    locally_biased : bool, optional
+        Whether to use the original [1]_ or modified [2]_ DIRECT algorithm.
+        Default is True. Possible values:
+
+        * ``locally_biased=False`` - use the original DIRECT algorithm
+        * ``locally_biased=True`` - use the modified DIRECT-l algorithm
+    f_min : float, optional
+        Function value of the global optimum.
+        By default it is a negative value whose absolute value is very large.
+        Set this value only if the global optimum is known.
+    f_min_per : float, optional
+        Terminate the optimization when the percent error satisfies:
+
+        .. math::
+
+            100*(f_{min} - f_{global})/\max(1, |f_{global}|) \leq f_{glper}
+
+        Default is 0.01.
+    vol_per : float, optional
+        Terminate the optimization once the volume of a hyperrectangle is less
+        than vol_per percent of the original hyper-rectangle.
+        Default is -1.
+    sigma_per : float, optional
+        Terminate the optimization once the measure of the hyperrectangle is
+        less than this argument. Default is -1.
+    callback : callable, `callback(xk)`, optional
+        A function to follow the progress of the minimization. ``xk`` is
+        the best solution found so far.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``success`` a
+        Boolean flag indicating if the optimizer exited successfully and
+        ``message`` which describes the cause of the termination. See
+        `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+
+    DIRECT is a deterministic
+    optimization algorithm capable of minimizing black box function with
+    its variables subject to lower and upper bound constrains by sampling
+    potential solutions in the search space. The algorithm starts with
+    mapping the hyperrectangle (set of all possible values that can be taken
+    by the input variables subject to the bound constraints) n-dimensional
+    unit hypercube. It samples the function at the center of this hypercube
+    and at 2n (n is the number of variables) more points, 2 in each coordinate
+    direction. Using these function values, DIRECT then divides the domain
+    into hyperrectangles, each having exactly one of the sampling points as
+    its center. In each iteration, DIRECT chooses, using the epsilon parameter,
+    by default 1e-4, some of the existing hyperrectangles to be further divided
+    . This division process continues until the maximum iterations or maximum
+    function evaluations allowed are exceeded, or the function value is within
+    the desired percentage error of the global minimum (if known). The improved
+    version of DIRECT algorithm [1]_ is biased towards local search making it
```

Comment:
```suggestion
    version of the DIRECT algorithm [1]_ is biased towards local search making it
```

Reply:
I don't understand `is biased towards local search`.
And isn't the improved version the reference `[2]`?

Reply:
N.B. https://link.springer.com/article/10.1023/A:1017930332101

I will update the reference.